pandas/io/excel.py

```diff
@@ -1610,35 +1672,57 @@ def _convert_to_style(self, style_dict, num_format_str=None):
             return None
 
         # Create a XlsxWriter format object.
-        xl_format = self.book.add_format()
+        props = {}
 
         if num_format_str is not None:
-            xl_format.set_num_format(num_format_str)
+            props['num_format'] = num_format_str
 
```

Comment:
ideally all of this logic would just be a single call here and the logic elsewhere.